quarkchain/cluster/shard_state.py

```diff
@@ -1281,6 +1287,14 @@ def add_cross_shard_tx_list_by_minor_block_hash(
         - it is a neighor of current shard following our routing rule
         """
         self.db.put_minor_block_xshard_tx_list(h, tx_list)
+        for tx in tx_list.tx_list:
+            asyncio.ensure_future(
+                self.subscription_manager.notify(
+                    "newPendingTransactions",
```

Comment:
prefer using a constant as in `quarkchain/constants.py`. the similar applies to the `SubscriptionManager`

Reply:
also `add_cross_shard_tx_list_by_minor_block_hash` seems to be invoked in 2 different cases

1. broadcast xshard tx from current shard to other shards
2. handle broadcasted xshard tx from other shards

and seems we only need to care about the second case, because the first case is already handled by `add_tx` for notifying pending tx

Reply:
Actually the pending tx response will not display twice as mentioned above empirically. For some reason (trying to figure out why), the subscription manager processed here is always target shard's, so only subscription of target shard will get notified.